azurerm/resource_arm_batch_pool.go

```diff
@@ -583,3 +626,43 @@ func validateUserIdentity(userIdentity *batch.UserIdentity) error {
 
 	return nil
 }
+
+func validateCrossFieldRules(pool *batch.Pool) error {
```

Comment:
in Go private functions are shared across the entire package; as such could we prefix this: